analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCallInsecureSecurityAlgorithm.cs

```diff
@@ -57,19 +57,17 @@ private void CheckInvocation(SyntaxNodeAnalysisContext context)
                 return;
             }
 
-            if (methodSymbol.ReturnType.DerivesFromAny(AlgorithmTypes) ||
-                IsInsecureBaseAlgorithmCreationFactoryCall(methodSymbol, invocation.ArgumentList))
+            if (methodSymbol.ReturnType.DerivesFromAny(AlgorithmTypes)
+                || IsInsecureBaseAlgorithmCreationFactoryCall(methodSymbol, invocation.ArgumentList))
             {
-                reportAllDiagnostics(context, invocation.GetLocation());
+                ReportAllDiagnostics(context, invocation.GetLocation());
             }
         }
 
         private bool IsInsecureBaseAlgorithmCreationFactoryCall(IMethodSymbol methodSymbol,
             ArgumentListSyntax argumentList)
         {
-            if (argumentList == null ||
-                methodSymbol?.ContainingType == null ||
-                methodSymbol.Name == null)
+            if (argumentList == null || methodSymbol?.ContainingType == null || methodSymbol.Name == null)
```

Comment:
To improve the coverage:
* When symbol exists, symbol name cannot be null. https://github.com/dotnet/roslyn/blob/57da48716decb252aac7f04499a1687429846151/src/Compilers/CSharp/Portable/Symbols/Symbol.cs#L63
* As line 60 declares, `methodSymbol` itself is never null at this path => we can remove the `?` to improve condition coverage. 

That should get the coverage green.

```suggestion
            if (argumentList == null || methodSymbol.ContainingType == null)
```